### ISSUE TITLE
Enhance KAN

### ISSUE THREAD

**Issue by DavidRamosArchilla** [ArnauMiro/pyLowOrder#76]
New functionalities will be added to the NN KAN model. Those are:

- Custom learning rate scheduling.
- The optimizer now can be changed. By default it uses Adam.
- More logging info is gathered, like learning rate evolution during training. 
- Now is easier to add more parameters to optimize by optima in terms of implementation.

**Comment by ArnauMiro** [comment]
@DavidRamosArchilla is this ready? You should add either me or Benet as reviewers so we can move forward your pull request.

**Comment by DavidRamosArchilla** [comment]
The test did not pass and it seems  to be related with the psutils library, i thought it was on the python standard library. I will create a new pr to solve this

**Comment by ArnauMiro** [comment]
Just solve it here

**Comment by ArnauMiro** [comment]
Now I got it, yes please create a new PR
